framework/aster-frame/src/task/processor.rs

```diff
@@ -117,9 +119,39 @@ fn switch_to_task(next_task: Arc<Task>) {
 
     let next_task_cx_ptr = &next_task.inner_ctx() as *const TaskContext;
 
+    // If the current and next tasks meet the following conditions,
+    // then we can safely skip switching page table for improved performance.
+    //
+    // Current task         | Next task                 | Should switch page table?
+    // ----------------------------------------------------------------------------
+    // any                  | w/o a user space          | No
+    // w/ a user space.     | w/ the same user space    | No
+    //
+    // In the table above, w/ is short for with and w/o is for without.
+    let should_switch_mm = match next_task.user_space() {
+        None => false,
+        Some(next_user_space) => {
+            let mut switch_mm = true;
+            if let Some(ref cur_task) = current_task {
+                if cur_task
+                    .user_space()
+                    .is_some_and(|cur_userspace| Arc::ptr_eq(cur_userspace, next_user_space))
+                {
+                    switch_mm = false;
+                }
+            }
+            switch_mm
+        }
+    };
+
     // change the current task to the next task
     PROCESSOR.lock().current = Some(next_task.clone());
+
     unsafe {
+        if should_switch_mm {
+            let user_space = next_task.user_space().unwrap();
+            user_space.vm_space().activate();
+        }
         context_switch(current_task_cx_ptr, next_task_cx_ptr);
```

Comment:
The `context_switch` function must never return when the previous task exits. So `current_task` and `next_task` must be dropped before calling `context_switch` to avoid resource leakage.

Ignoring this comment is pretty much fine, as I will do when I rebase #821.

Actually, I've added comments in #821 to emphasize this point. Have a look at https://github.com/lrh2000/asterinas/commit/fe84db307f138828e06631c70e4cfedcae2e3dd0#diff-31b4cc1d8756834466e8c9d51c750d62acae54b484bd20669e4891245ddccff1L132-R146

Reply:
Agreed, it would be more suitable to address this issue in #821.